src/app/App.js

```diff
@@ -284,8 +284,13 @@ class App extends EventEmitter {
 	 * @return {boolean}
 	 */
 	canNavigate(url) {
-		var path = utils.getUrlPath(url);
-		var uri = new Uri(url);
+		const uri = utils.validateUrl(url);
```

Comment:
What if we return the `Uri` instance instead of return it inside the `result` when no error occur? 

It will make easier to access its attributes.